resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/localstore/FileSystemResourceManager.java

```diff
@@ -1099,13 +1099,14 @@ public void updateLocalSync(ResourceInfo info, long localSyncInfo) {
 	}
 
 	/**
-	 * The target must exist in the workspace. The content InputStream is
-	 * closed even if the method fails. If the force flag is false we only write
-	 * the file if it does not exist or if it is already local and the timestamp
-	 * has NOT changed since last synchronization, otherwise a CoreException
-	 * is thrown.
+	 * The target must exist in the workspace and must remain existing while
+	 * executing this method. The content InputStream is closed even if the method
+	 * fails. If the force flag is false we only write the file if it does not exist
+	 * or if it is already local and the timestamp has NOT changed since last
+	 * synchronization, otherwise a CoreException is thrown.
 	 */
 	public void write(IFile target, InputStream content, IFileInfo fileInfo, int updateFlags, boolean append, IProgressMonitor monitor) throws CoreException {
+		assertExists(target);
```

Comment:
that violates the javadoc "the content InputStream is closed even if the method	fails" please put into the try block

Reply:
fixed in amended commit f225d48